### PR TITLE
Update event.js manage passive event listener

### PR DESCRIPTION
**Pull request by SoraWebsoft**
### Summary ###
Correct version of manage passive event Listener to correct Best Practice for Google / Lighthouse
See video for view impact on web page : https://www.youtube.com/watch?v=NPM6172J22g

Issue  #3804 - Non-passive even listener violation

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works

### CODE REVIEW

**Comment by mgol** [comment]
Thanks for the PR but this is not so simple. We need to support non-passive event listeners as well. Fixing the issue will require a significant refactor of the whole event subsystem in jQuery which is planned for 4.0.0 & tracked in #2871.

**Comment by SoraWebsoft** [comment]
> Thanks for the PR but this is not so simple. We need to support non-passive event listeners as well. Fixing the issue will require a significant refactor of the whole event subsystem in jQuery which is planned for 4.0.0 & tracked in #2871.

I have tested this in my jQuery 3.5.0 version and it's work. 
That's why I wanted to share this patch.

Regards.

File: src/event.js [hunk 1]
```diff
@@ -14,7 +14,28 @@ import "./selector.js";
...
+			passiveSupported = true;
+
+			// If web browser manage passive listeners, then we pass in array with all events to put in passive listener
+			passiveListeners = [ "wheel", "mousewheel", "touchmove", "touchstart", "touchend" ];
+		}
+	} );
+	window.addEventListener( "test", options, options );
+	window.removeEventListener( "test", options, options );
+} catch ( err ) {
+	passiveSupported = false;
```
Comment (dmethvin) [inline src/event.js:35]: Do we need this since it's already initialized to `false`?

File: src/event.js [hunk 2]
```diff
@@ -14,7 +14,28 @@ import "./selector.js";
 var
 	rkeyEvent = /^key/,
 	rmouseEvent = /^(?:mouse|pointer|contextmenu|drag|drop)|click/,
-	rtypenamespace = /^([^.]*)(?:\.(.+)|)/;
+	rtypenamespace = /^([^.]*)(?:\.(.+)|)/,
+	passiveSupported = false,
+	passiveListeners = [],
+	options = null;
+
+// Passive Listener Check
```
Comment (dmethvin) [inline src/event.js:22]: Don't really need these comments.

File: src/event.js [hunk 3]
```diff
@@ -189,7 +210,21 @@ jQuery.event = {
...
-						elem.addEventListener( type, eventHandle );
+
+						// Passive Listener, If web browser manage passive listeners
+						if ( passiveSupported === true ) {
+
+							// If current listener (type) is in passive listener list
+							if ( passiveListeners.indexOf( type ) >= 0 ) {
+
+								// We declare it to passive listener {passive: true}
+								elem.addEventListener( type, eventHandle, { passive: true } );
```
[the remark this answers is not in the document]
Comment (mgol) [reply src/event.js:221]: Exactly. It's a frequent use case to call `preventDefault()` on a wheel event.

Also, touch event listeners are passive by default in latest Chrome, Firefox & Safari so for touch events this is not even needed.

**Pull request closed without merging**